tracer/test/test-applications/instrumentation/CallTargetNativeTest/With0Arguments.cs

```diff
@@ -1,41 +1,638 @@
+        public void VoidMethod() { }
+        public int ReturnValueMethod() => 42;
+        public string ReturnReferenceMethod() => "Hello World";
+        public T ReturnGenericMethod<T>() => default;
+    }
+    public class With0ArgumentsGeneric<T>
+    {
+        public void VoidMethod() { }
+        public int ReturnValueMethod() => 42;
+        public string ReturnReferenceMethod() => "Hello World";
+        public T ReturnGenericMethod() => default;
+    }
+    public class With0ArgumentsInherits : With0Arguments { }
+    public class With0ArgumentsInheritsGeneric : With0ArgumentsGeneric<int> { }
+    public struct With0ArgumentsStruct
+    {
+        public void VoidMethod() { }
+        public int ReturnValueMethod() => 42;
+        public string ReturnReferenceMethod() => "Hello World";
+        public T ReturnGenericMethod<T>() => default;
+    }
+    public static class With0ArgumentsStatic
+    {
+        public static void VoidMethod() { }
+        public static int ReturnValueMethod() => 42;
+        public static string ReturnReferenceMethod() => "Hello World";
+        public static T ReturnGenericMethod<T>() => default;
+    }
+    public class With0ArgumentsThrowOnBegin : With0Arguments { }
+    public class With0ArgumentsThrowOnEnd : With0Arguments { }
+    public class With0ArgumentsThrowOnAsyncEnd
+    {
+        public Task Wait2Seconds() => Task.Delay(2000);
+    }
+}
+
+partial struct ArgumentsStructParentType
+{
+    // *** With0Arguments
+    public class With0Arguments
+    {
+        public void VoidMethod() { }
+        public int ReturnValueMethod() => 42;
+        public string ReturnReferenceMethod() => "Hello World";
+        public T ReturnGenericMethod<T>() => default;
+    }
+    public class With0ArgumentsGeneric<T>
+    {
+        public void VoidMethod() { }
+        public int ReturnValueMethod() => 42;
+        public string ReturnReferenceMethod() => "Hello World";
+        public T ReturnGenericMethod() => default;
+    }
+    public class With0ArgumentsInherits : With0Arguments { }
+    public class With0ArgumentsInheritsGeneric : With0ArgumentsGeneric<int> { }
+    public struct With0ArgumentsStruct
+    {
+        public void VoidMethod() { }
+        public int ReturnValueMethod() => 42;
+        public string ReturnReferenceMethod() => "Hello World";
+        public T ReturnGenericMethod<T>() => default;
+    }
+    public static class With0ArgumentsStatic
+    {
+        public static void VoidMethod() { }
+        public static int ReturnValueMethod() => 42;
+        public static string ReturnReferenceMethod() => "Hello World";
+        public static T ReturnGenericMethod<T>() => default;
+    }
+    public class With0ArgumentsThrowOnBegin : With0Arguments { }
+    public class With0ArgumentsThrowOnEnd : With0Arguments { }
+    public class With0ArgumentsThrowOnAsyncEnd
+    {
+        public Task Wait2Seconds() => Task.Delay(2000);
+    }
+}
+
+partial class ArgumentsGenericParentType<PType>
```

Comment:
I wonder what you think: even though we cannot instrument generic structs, should we create all of the valid combinations of inner types here and just comment out code that would call these types? This would be similar to the current status of the Derived tests. That way we can "document" here a valid user scenario that we cannot support? You've already written a good comment within the CallTarget native code but maybe this could help slightly

Reply:
I did that with `Derived tests` because we can enable support in a future PR so it's just a matter of removing comments to enable the tests.

For `generic structs` I think we never going to support it, the reason is because we need to emit the `ldobj` instruction with the concrete generic type (typespec) because that one is the one that contains the exact size of the struct. 

> The number of bytes copied depends on the size of the class (as specified by the class parameter). The class parameter is a metadata token representing the value type.

https://docs.microsoft.com/en-us/dotnet/api/system.reflection.emit.opcodes.ldobj?view=net-6.0

The problem is that at rewrite time we only have the `__Cannon` type :(

I don't know if live debugger folks have an amazing solution for this case though...

Reply:
Unfortunately for the purpose you describe (`ldobj` with concrete `typespec` in case of getting the parent of generic struct method), we too had no real solution to it.